helm-charts/splunk-otel-collector/templates/_helpers.tpl

```diff
@@ -149,7 +149,8 @@ Define name for the Splunk Secret
 {{- if .Values.secret.name -}}
 {{- printf "%s" .Values.secret.name -}}
 {{- else -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
```

Comment:
@VihasMakwana my comment [here](https://github.com/signalfx/splunk-otel-collector-chart/pull/873#discussion_r1287273674) is still relevant wrt the new changes. The helper `splunk-otel-collector.fullname` will make certain the secret name aligns correctly with the release name and the if/else can be dropped in favor of using the `default` function.

Reply:
I'm not sure if we can directly use `splunk-otel-collector.fullname`, as it takes `.Values.fullnameOverride` into account. We don't have any reference to `.Values.fullnameOverride` in the existing logic of `splunk-otel-collector.secret`.
What we can do is separate out [this block](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/_helpers.tpl#L19-L23 ) into a separate helper and use that instead.

Reply:
But this change would be more like a refactoring, rather than a bug fix.
I would like to keep it in separate PR, if that's fine by you.

Reply:
The `fullnameOverride` variable is meant to be used when a user wants to intentionally replace the complete name of the objects generated as part of the chart. We use the `fullname` helper function to name all other objects already and this gives user the option to change the names of the objects in a consistent manner across the board. So I don't think we are changing the scope of the original issue by using the fullname helper here.